lms/djangoapps/discussion_api/api.py

```diff
@@ -43,15 +43,15 @@
 from openedx.core.djangoapps.course_groups.cohorts import get_cohort_id
 
 
-def _get_course_or_404(course_key, user):
+def _get_course_or_raise_not_found(course_key, user):
     """
-    Get the course descriptor, raising Http404 if the course is not found,
+    Get the course descriptor, raising ObjectDoesNotExist if the course is not found,
     the user cannot access forums for the course, or the discussion tab is
     disabled for the course.
     """
     course = get_course_with_access(user, 'load', course_key, check_if_enrolled=True)
     if not any([tab.type == 'discussion' and tab.is_enabled(course, user) for tab in course.tabs]):
-        raise Http404
+        raise ObjectDoesNotExist("Course not found.")
```

Comment:
Instead of raising a generic exception type, can you raise an exception specific to the Course Not being Found?
In fact, there's an existing [CourseNotFoundError](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/course_structures/api/v0/errors.py#L4) that we should move to a more common place and have it inherit from ObjectDoesNotExist.  So:

1. Move `CourseNotFoundError` to openedx/core/lib/exceptions.py (I can't believe we've survived this long without this exception).
2. Have `CourseNotFoundError` inherit from `ObjectDoesNotExist` so that your changes in viewutils.py still apply.